### PR TITLE
Add empty folder export, folder comments, and / escaping

### PR DESCRIPTION
**Pull request by AndrianBdn**
closes https://github.com/Hypersequent/tms-issues/issues/2215

Implements features from Hypersequent/tms-issues#2215 (partially covered by #5, which had review feedback that is addressed here):

- **Empty folder export** — `AddFolder(folder, comment)` API to create folders with no test cases. Folder-only rows are written when a folder is empty or has a comment.
- **Folder comments** — New `"Folder Comment"` CSV column. Comments are set via `AddFolder` and rendered on a dedicated folder row — **this feature has been tested on staging of 26W06**
- **`/` in folder names** — Removed `excludesall=/` validation. `/` in folder segments is escaped as `\/` in the joined path. Folder segments ending with `\` are rejected to prevent ambiguity.
- **`jsonMarshal` with `SetEscapeHTML(false)`** — Prevents Go's default `json.Marshal` from mangling `&`, `<`, `>` into `\u0026`, `\u003c`, `\u003e` in JSON fields.
- **Insertion-order folders** — Folders appear in CSV in the order they were added, not alphabetically sorted.

## PR #5 review feedback addressed

| satvik007 comment | Status |
|---|---|
| Folder escaping is dead code due to `excludesall=/` validation | Fixed — removed validation, escaping is now active |
| "You also need to trim stray `\` at the end of folder name" | Added — segments ending with `\` are rejected |
| CSV row uses `strings.Join(tc.Folder, "/")` instead of escaped map key | Fixed — uses escaped key from `folderTCaseMap` |
| `jsonMarshal` newline: "Is there some TrimSuffix function for this" | Fixed — uses `bytes.TrimSuffix` |
| Folder sorting is alphabetical vs DB position order | Fixed — preserves insertion order |

## Test plan

- [x] All existing tests updated and passing
- [x] New unit tests for: slash escaping, backslash rejection, empty folders, folder comments, folder with comment + test cases, AddFolder validation, slash in folder names
- [x] Manual import verification using `go run examples/functional/main.go` (generates `folder_comments.csv`, `empty_folders.csv`, `escaping.csv`)

### CODE REVIEW

File: qascsv.go [hunk 1]
```diff
@@ -287,12 +343,17 @@ func (q *QASphereCSV) addTCase(tc TestCase) {
...
+	var buf bytes.Buffer
+	enc := json.NewEncoder(&buf)
+	enc.SetEscapeHTML(false)
+	if err := enc.Encode(v); err != nil {
+		return nil, err
 	}
-	slices.Sort(folders)
-	return folders
+	return bytes.TrimSuffix(buf.Bytes(), []byte("\n")), nil
 }
```
[the remark this answers is not in the document]
Comment (AndrianBdn) [reply qascsv.go:357]: data in this columns are decoded; not inserted in HTML

File: qascsv.go [hunk 2]
```diff
@@ -235,6 +238,35 @@ func (q *QASphereCSV) AddTestCases(tcs []TestCase) error {
...
+		return errors.Errorf("folder %q already exists", folderPath)
+	}
+
+	q.folderOrder = append(q.folderOrder, folderPath)
+	q.folderTCaseMap[folderPath] = nil
+	if comment != "" {
+		q.folderCommentMap[folderPath] = comment
+	}
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (AndrianBdn) [reply qascsv.go]: go's CSV package does this

**Review by sirkostya009 — APPROVED** [review]

File: examples/functional/main.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	generateFolderComments()
+	generateEmptyFolders()
+	generateEscaping()
+}
+
+func generateFolderComments() {
+	q := qascsv.NewQASphereCSV()
+
+	if err := q.AddFolder([]string{"Commented Folder"}, "This folder has a comment but also contains test cases"); err != nil {
+		log.Fatal(err)
```
Comment (satvik007) [inline examples/functional/main.go]: Instead of so many log.Fatal its better to return the error and handle at caller.

File: examples/functional/main.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+	}
+	if err := q.AddTestCase(qascsv.TestCase{
+		Title:    "Test inside commented folder",
+		Folder:   []string{"Commented Folder"},
```
Comment (satvik007) [inline examples/functional/main.go]: We could call this FolderPath for more clarity

File: qascsv.go [hunk 5]
```diff
@@ -235,6 +238,35 @@ func (q *QASphereCSV) AddTestCases(tcs []TestCase) error {
...
+		if seg == "" {
+			return errors.New("folder segment must not be empty")
+		}
+		if len(seg) > 255 {
+			return errors.Errorf("folder segment %q exceeds 255 character limit", seg)
+		}
+	}
+	if err := validateFolderSegments(folder); err != nil {
+		return err
+	}
```
Comment (satvik007) [inline qascsv.go]: missed opportunity to move validation in validateFolderSegments.

File: qascsv.go [hunk 6]
```diff
@@ -235,6 +238,35 @@ func (q *QASphereCSV) AddTestCases(tcs []TestCase) error {
 	return nil
 }
 
+func (q *QASphereCSV) AddFolder(folder []string, comment string) error {
+	if len(folder) == 0 {
+		return errors.New("folder path must not be empty")
+	}
+	for _, seg := range folder {
```
Comment (satvik007) [inline qascsv.go]: if we use input struct, that improves the code
- makes the arguments more clear in example code
- we can use go-validator validation consistently for both tcase and here.

**Review by satvik007 — CHANGES_REQUESTED** [review]

File: examples/functional/main.go (hunk 4 above)
Comment (AndrianBdn) [reply examples/functional/main.go]: for throwaway example this is a bit drastic, but okay

File: qascsv.go [hunk 7]
```diff
@@ -113,7 +121,7 @@ type TestCase struct {
 	// for reference. (optional)
 	LegacyID string `validate:"max=255"`
 	// The complete folder path to the test case. (required)
-	Folder []string `validate:"min=1,dive,required,max=255,excludesall=/"`
+	Folder []string `validate:"min=1,dive,required,max=255"`
```
Comment (satvik007) [inline qascsv.go]: This should also be renamed to FolderPath, it will make the purpose of the field more clear.

**Review by satvik007 — APPROVED** [review]
